assessInterruptedSession hides stale crash locks with matching PID, preventing startAuto cleanup

## Problem

When a previous auto-mode session crashes and leaves a stale `auto.lock`, `startAuto` fails to clear it if the current process has the same PID as the dead session. This leaves a misleading crash artifact that shows up in `gsd doctor` output and can block subsequent auto-mode starts.

## Root Cause

`interrupted-session.js:68` filters out crash locks whose PID matches the current process:

```javascript
const rawLock = readCrashLock(basePath);
const lock = rawLock && rawLock.pid !== process.pid ? rawLock : null;
```

`startAuto` (`auto.js`) only calls `clearLock(base)` when `freshStartAssessment.lock` is truthy:

```javascript
if (freshStartAssessment.lock) {
    emitCrashRecoveredUnitEnd(base, freshStartAssessment.lock);
    clearLock(base);
}
```

Because `assessInterruptedSession` returns `lock: null` for same-PID locks, `startAuto` never clears them. The stale lock persists until the user manually runs `/gsd doctor --fix` or deletes `.gsd/auto.lock`.

This is especially problematic when the extension reloads in the same Node process (same PID) or when the OS reuses a PID quickly.

## Expected Behavior

`assessInterruptedSession` should return the lock even when the PID matches, but classify it as `stale` (not `running`). `startAuto` should then clear the lock and emit a synthetic `unit-end` if needed.

Alternatively, `startAuto` should unconditionally clear any lock whose PID is dead, regardless of `freshStartAssessment.lock`.

## Environment

- GSD version: 2.81.0
- Stale lock: PID 76021, unit `execute-task M011/S07/T05`, started 2026-05-09T04:21:57.519Z
- Doctor reports `stale_crash_lock` with `fixable: true`, but `/gsd auto` startup skips cleanup because `assessInterruptedSession` returns `lock: null`.

## Forensic Evidence

From the forensic report for this session:
- `[ERROR] doctor-issue`: Doctor: Stale auto-mode worker (PID 76021, started 2026-05-09T04:21:57.519Z, was executing execute-task M011/S07/T05) — process is no longer running
- The session continued to M012 without clearing the stale lock artifact.
- `readCrashLock(basePath)` found the lock, but `assessInterruptedSession` discarded it due to PID match.

## Fix suggestion

In `interrupted-session.js:68`, change the logic to preserve the lock when the PID is dead, even if it matches the current process:

```javascript
const rawLock = readCrashLock(basePath);
const lock = rawLock && !isLockProcessAlive(rawLock) ? rawLock : null;
```

Then let `startAuto` handle same-PID locks the same way it handles other stale locks.

---
*Auto-generated by `/gsd forensics`*

--- a/src/resources/extensions/gsd/interrupted-session.ts
+++ b/src/resources/extensions/gsd/interrupted-session.ts
@@ -128,8 +128,7 @@ export async function assessInterruptedSession(
     ? existsSync(pausedSession.worktreePath)
     : false;
   const assessmentBasePath = worktreeExists ? pausedSession!.worktreePath! : basePath;
-  const rawLock = readCrashLock(basePath);
-  const lock = rawLock && rawLock.pid !== process.pid ? rawLock : null;
+  const lock = readCrashLock(basePath);
 
   if (!lock && !pausedSession) {
     return {
@@ -146,7 +145,7 @@ export async function assessInterruptedSession(
     };
   }
 
-  if (lock && isLockProcessAlive(lock)) {
+  if (lock && lock.pid !== process.pid && isLockProcessAlive(lock)) {
     return {
       classification: "running",
       lock,

--- a/src/resources/extensions/gsd/tests/interrupted-session-auto.test.ts
+++ b/src/resources/extensions/gsd/tests/interrupted-session-auto.test.ts
@@ -226,6 +226,14 @@ test("direct /gsd auto skips paused-session replay when recovered unit already c
   }
 });
 
+test("interrupted-session source preserves raw lock and excludes same-pid from running classification", async () => {
+  const source = await import(`node:fs/promises`).then((fs) =>
+    fs.readFile(new URL("../interrupted-session.ts", import.meta.url), "utf-8")
+  );
+  assert.ok(source.includes('const lock = readCrashLock(basePath);'));
+  assert.ok(source.includes('if (lock && lock.pid !== process.pid && isLockProcessAlive(lock)) {'));
+});
+
 test("auto module imports successfully after interrupted-session changes", async () => {
   const mod = await import(`../auto.ts?ts=${Date.now()}-${Math.random()}`);
   assert.equal(typeof mod.startAuto, "function");

